homework/index.js

```diff
@@ -28,26 +30,133 @@
     });
     return elem;
   }
+  function formatDate(dateString) {
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderError(error) {
+    const container = document.getElementById('main-container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'alert',
+    });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
+  function addTable(table, body, value) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, { text: body });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoDetails(repo) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const ul = createAndAppend('ul', leftSide);
+    const li = createAndAppend('li', ul);
+    const table = createAndAppend('table', li);
+    const tr1 = addTable(table, 'Repository:', '');
+    createAndAppend('a', tr1.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+    addTable(table, 'Description:', repo.description);
+    addTable(table, 'Fork: ', repo.forks);
+    addTable(table, 'Updated:', formatDate(repo.updated_at));
+  }
+
+  function fetchAndRender(repo) {
```

Comment:
This name `fetchAndRender` (and the name `startPage` below) remind me of the homework of previous classes 🤔.